codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,253 @@
+            "MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk. " +
+                    "Commands that may modify the data set are disabled. Please check Redis logs for details " +
+                    "about the error.",
+            "NOAUTH Authentication required.",
+            "NOREPLICAS Not enough good slaves to write.",
+            "NOSCRIPT No matching script. Please use EVAL.",
+            "OOM command not allowed when used memory > 'maxmemory'.",
+            "READONLY You can't write against a read only slave.",
+            "WRONGTYPE Operation against a key holding the wrong kind of value",
+    };
+
+    private static PreloadStringsCache preloadSimpleString;
+    private static PreloadStringsCache preloadErrors;
+
+    static {
+        setPreloadStrings(RedisMessageType.SIMPLE_STRING, DEFAULT_PRELOAD_SIMPLE_STRINGS);
+        setPreloadStrings(RedisMessageType.ERROR, DEFAULT_PRELOAD_ERRORS);
+    }
+
+    private RedisMessageFactory() {
+    }
+
+    /**
+     * Set preload strings. Users can override preload strings before using {@link RedisMessageFactory}.
+     *
+     * @param type one of {@link RedisMessageType#SIMPLE_STRING} or {@link RedisMessageType#ERROR}.
+     * @param messages an array of String to preload.
+     */
+    public static void setPreloadStrings(RedisMessageType type, String... messages) {
+        ObjectUtil.checkNotNull(messages, "messages");
+        if (type == RedisMessageType.SIMPLE_STRING) {
+            preloadSimpleString = convertPreloadsCache(RedisMessageType.SIMPLE_STRING, messages);
+        } else if (type == RedisMessageType.ERROR) {
+            preloadErrors = convertPreloadsCache(RedisMessageType.ERROR, messages);
+        } else {
+            throw new IllegalArgumentException("not support type: " + type);
+        }
+    }
+
+    private static PreloadStringsCache convertPreloadsCache(RedisMessageType type, String... messages) {
+        Map<String, RedisMessage> stringsCache = new HashMap<String, RedisMessage>();
+        Map<ByteBuf, RedisMessage> byteBufCache = new HashMap<ByteBuf, RedisMessage>();
+        for (String message : messages) {
+            ByteBuf byteBuf = Unpooled.wrappedBuffer(message.getBytes(CharsetUtil.UTF_8));
+            RedisMessage preloadMessage = new StringRedisMessage(type, message);
+            stringsCache.put(message, preloadMessage);
+            byteBufCache.put(byteBuf, preloadMessage);
+        }
+        return new PreloadStringsCache(stringsCache, byteBufCache);
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#SIMPLE_STRING}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createSimpleString(String content) {
+        ObjectUtil.checkNotNull(content, "content");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "content" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1192)